src/modules/CC1101/CC1101.cpp

```diff
@@ -218,24 +213,54 @@ int16_t CC1101::startTransmit(uint8_t* data, size_t len, uint8_t addr) {
   int16_t state = SPIsetRegValue(CC1101_REG_IOCFG0, CC1101_GDOX_SYNC_WORD_SENT_OR_RECEIVED);
   RADIOLIB_ASSERT(state);
 
+  // data put on FIFO.
+  uint8_t dataSent = 0;
+
   // optionally write packet length
   if (_packetLengthConfig == CC1101_LENGTH_CONFIG_VARIABLE) {
+
+    // enforce variable len limit.
+    if (len > 254) {
+      return (ERR_PACKET_TOO_LONG);
+    }
+
     SPIwriteRegister(CC1101_REG_FIFO, len);
+    dataSent += 1;
   }
 
   // check address filtering
   uint8_t filter = SPIgetRegValue(CC1101_REG_PKTCTRL1, 1, 0);
   if(filter != CC1101_ADR_CHK_NONE) {
     SPIwriteRegister(CC1101_REG_FIFO, addr);
+    dataSent += 1;
   }
 
-  // write packet to FIFO
-  SPIwriteRegisterBurst(CC1101_REG_FIFO, data, len);
+  // fill the FIFO.
+  uint8_t initialWrite = min(len, (CC1101_FIFO_SIZE - dataSent));
+  SPIwriteRegisterBurst(CC1101_REG_FIFO, data, initialWrite);
+  dataSent += initialWrite;
 
   // set mode to transmit
   SPIsendCommand(CC1101_CMD_TX);
 
-  return(state);
+  // keep feeding the FIFO until the packet is over.
+  uint8_t bytesInFIFO;
+  while (dataSent < len) {
+    // get number of bytes in FIFO.
+    bytesInFIFO = SPIgetRegValue(CC1101_REG_TXBYTES, 6, 0);
+
+    // if there's room then put other data.
+    if (bytesInFIFO < CC1101_FIFO_SIZE) {
+      uint8_t bytesToWrite = min(CC1101_FIFO_SIZE - bytesInFIFO, len - dataSent);
+      SPIwriteRegisterBurst(CC1101_REG_FIFO, &data[dataSent], bytesToWrite);
+      dataSent += bytesToWrite;
+    } else {
+      // wait for radio to send some data.
+      delay(1);
```

Comment:
That's a good question,
I really don't know;  Currently I don't have two boards to test at the max speed, I'm using it with a key fob at very low dr.

Maybe is it the case to lower it down?

Reply:
Seems like it won't be an issue (had it running with 250 byte packets at 120 kbps, 100 kHz frequency deviation, 464 kHz Rx bw).

However, I did notice that upon starting the receiver (i.e. uploading example sketch), it initially returns a bunch of CRC errors, before getting valid packets. The receiver CRC errors are generated only when packet is sent (so it's not just some random noise or other transmissions). It seems to happen regardless of packet length or modem configuration and even happens using unmodified examples. Have you seen that behavior before?

Reply:
That's odd.
You mean that when you send from a RadioLib instance to another, the receiving one gets a CRC error? This means that either `readData()` or `trasmit()` implementation is buggy.
Did you try if it behaves the same way with the 'old' code (ex: old TX -> new RX) ?

I've tested the demo sketches but patched them adding:
```cpp
  state |= cc.setOOK(true);
  state |= cc.setCrcFiltering(false);
  state |= cc.fixedPacketLengthMode(78);
  byte syncWord[2] = {0x99, 0x9A};
  state |= cc.setSyncWord(syncWord, 2);
```
I disabled CRC since I'm transmitting from a key fob with unknown CRC function.
That's why I missed this error. :sob: 

I'll manage to get two CC1101 and test this out.

Reply:
> You mean that when you send from a RadioLib instance to another, the receiving one gets a CRC error

That's basically what's happening. Usually after 10-20 packets the CRC errors seem to disappear, and normal operations resumes, but sometimes it can take up to several minutes. On reset, the CRC errors are back.

This only happens when the receiver is running the code from this PR, it doesn't happen with the old implementation (or when CRC filtering is disabled).

Reply:
How to proceed?
Maybe it's better to close this PR, I'll submit this again if I find a way to fix it.

Reply:
That's probably for the best, unfortunately I don't have the time right now to investigate this.